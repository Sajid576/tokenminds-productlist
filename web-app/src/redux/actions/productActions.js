import fakestoreapi from "../../apis/fakeStoreApi";
import { ActionTypes } from "../constants/action-types";
import { callGetApi } from "../../lib/api";

export const fetchProducts = () => async (dispatch) => {
  const response = await callGetApi("http://localhost:7000/api/product");
  // const response = await fakestoreapi.get('/products');
  console.debug("response", response);
  dispatch({ type: ActionTypes.FETCH_PRODUCTS, payload: response.data });
};

export const fetchProduct = (id) => async (dispatch) => {
  // const response = await fakestoreapi.get(`/products/${id}`);
  const response = await callGetApi(`http://localhost:7000/api/product/${id}`);
  dispatch({ type: ActionTypes.SELECTED_PRODUCT, payload: response.data });
};

export const setProducts = (products) => {
  return {
    type: ActionTypes.SET_PRODUCTS,
    payload: products,
  };
};

export const selectedProduct = (product) => {
  return {
    type: ActionTypes.SELECTED_PRODUCT,
    payload: product,
  };
};

export const removeSelectedProduct = () => {
  return {
    type: ActionTypes.REMOVE_SELECTED_PRODUCT,
  };
};
