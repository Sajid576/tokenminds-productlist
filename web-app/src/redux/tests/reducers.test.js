import { fetchProducts } from '../actions/productActions';
import { productReducer } from '../reducers/productReducer';

describe('product reducers', () => {
  it('returns initial state', () => {
    expect(productReducer(undefined, {})).toMatchSnapshot();
  });

  it('Handle fetchProducts action', () => {
    expect(productReducer(undefined, fetchProducts())).toMatchSnapshot();
  });
});
