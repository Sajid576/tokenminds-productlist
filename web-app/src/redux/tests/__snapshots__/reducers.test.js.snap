// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product reducers Handle fetchProducts action 1`] = `
Object {
  "products": Array [],
}
`;

exports[`product reducers returns initial state 1`] = `
Object {
  "products": Array [],
}
`;
